### PR TITLE
Jacobian Slice Bug StrawberryFields

### PR DESCRIPTION
**Pull request by puzzleshark**

### CODE REVIEW

File: pennylane/interfaces/autograd.py [hunk 1]
```diff
@@ -229,7 +229,15 @@ def grad_fn(dy):
...
 
-        return [qml.math.to_numpy(v, max_depth=_n) if isinstance(v, ArrayBox) else v for v in vjps]
+        return_vjps = [
+            qml.math.to_numpy(v, max_depth=_n) if isinstance(v, ArrayBox) else v for v in vjps
+        ]
+        if device.capabilities().get("provides_jacobian", False):
+            # in the case where the device provides the jacobian,
+            # the output of grad_fn must be wrapped in a tuple in
+            # order to match the input parameters to _execute.
+            return (return_vjps,)
```
Comment (antalszava) [inline pennylane/interfaces/autograd.py:239]: :partying_face: 

This is a great catch :clap:

File: tests/gradients/test_custom_jacobian.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import pennylane.numpy as np
+import pennylane as qml
+
+from pennylane.devices.default_qubit import DefaultQubit
+
+class CustomJacobianDevice(DefaultQubit):
```
Comment (antalszava) [inline tests/gradients/test_custom_jacobian.py]: This would best belong in the `tests/interfaces/autograd.py` file as it's an autograd interface specific test.

Comment (antalszava) [reply tests/gradients/test_custom_jacobian.py]: @puzzleshark also: don't forget to add an entry to the changelog

File: doc/releases/changelog-0.23.0.md [hunk 3]
```diff
@@ -614,12 +614,15 @@ the `decimals` and `show_matrices` keywords are added. `qml.drawer.tape_text(tap
   the qubit operator that is prepared for tapering the HF state.  
   [(#2441)](https://github.com/PennyLaneAI/pennylane/pull/2441)
 
+* Fixed a bug with custom device defined jacobians not being returned properly.
+  [(#108)](https://github.com/PennyLaneAI/pennylane-sf/pull/108/files)
```
Comment (antalszava) [inline doc/releases/changelog-0.23.0.md:618]:
```suggestion
  [(#2485)](https://github.com/PennyLaneAI/pennylane-sf/pull/2485)
```

**Review by antalszava — APPROVED** [review]
Looks good :+1: Great one tracking this down and fixing it! :muscle: :detective:

It's also nice to have the device jacobian case be supported

File: tests/interfaces/test_autograd.py [hunk 4]
```diff
@@ -1111,3 +1112,30 @@ def test_multiple_hamiltonians_trainable(self, cost_fn, execute_kwargs, tol):
...
+        def circuit(v):
+            qml.RX(v, wires=0)
+            return qml.probs(wires=[0, 1])
+
+        d_circuit = qml.jacobian(circuit, argnum=0)
+
+        params = np.array(1.0, requires_grad=True)
+
+        d_out = d_circuit(params)
+        assert np.allclose(d_out, np.array([1.0, 2.0, 3.0, 4.0]))
```
Comment (cvjjm) [inline tests/interfaces/test_autograd.py:1141]: @antalszava I am trying to understand why this bug didn't cause a problem for us. What was `d_out` before this fix? Thanks very much!

Comment (josh146) [reply tests/interfaces/test_autograd.py:1141]: This is a very good question! Another good question is if this change would affect your custom device jacobians 🤔

Comment (cvjjm) [reply tests/interfaces/test_autograd.py:1141]: Now I see: We had actually worked around this by returning `np.array([gradient])` from our device's jacobian method, where gradient is itself a numpy array. I guess we just need to stop doing this starting with PL v0.23.0 then. Thanks for the heads up!

File: doc/releases/changelog-0.23.0.md [hunk 5]
```diff
@@ -614,12 +614,15 @@ the `decimals` and `show_matrices` keywords are added. `qml.drawer.tape_text(tap
   the qubit operator that is prepared for tapering the HF state.  
   [(#2441)](https://github.com/PennyLaneAI/pennylane/pull/2441)
 
+* Fixed a bug with custom device defined jacobians not being returned properly.
+  [(#2485)](https://github.com/PennyLaneAI/pennylane-sf/pull/2485)
```
[the remark this answers is not in the document]
Comment (antalszava) [reply doc/releases/changelog-0.23.0.md:618]: Yes, thank you for catching that! :tada:
